### DISCUSSION TITLE
How to store data from server to client?

### DISCUSSION THREAD

**Discussion by jacob-sheldon** [vercel/next.js#53773] [Help]
### Summary

In my next.js app, I send a request to get a csrf token from the server while the app starts, this csrf token will be sent by every subsequent request as one of the headers. Now I can get this token from a server component, but I don't know how to deliver it to other requests.
The cookie and localstorage cannot be used on the server side, so the token can not be saved in them. 
I know something about redux, maybe it can be used in this situation, but I think it is too heavy right now for this project.
If you know some methods about this problem, please tell me how you used for it. Or I must use redux?

**Comment by devjiwonchoi** [comment]
Set token on cookies via `cookies()`.

### Reference
- [cookies](https://nextjs.org/docs/app/api-reference/functions/cookies)
